features/2025-08-14-terraform-settings.md

```diff
@@ -0,0 +1,554 @@
+3. Radius does not block the ability to use any Terraform features including `terraform plan`.
+4. Radius does not manage Terraform, it only executes the Terraform CLI.
+
+### Goals
+
+* Enable platform engineers to install and upgrade their preferred version of the Terraform binary independently of Radius
+* Enable platform engineers the ability to specify Recipe settings independently of Environments to ease managing Environments at scale (similar objectives of Recipe Packs)
+* Enable platform engineers to provide their existing Terraform settings to Radius without modification, including using and authenticating to provider mirrors and internal module sources
+* Provide trace-level logs for Terraform when executed by Radius
+
+### Non-Goals (out of scope)
+
+* `terraform plan` functionality (this will be addressed in a future feature specification)
+
+## Scenario 1 – Installing and Updating Terraform
+
+### User Story 1 – Install Terraform
+
+***As a platform engineer, I need to install Terraform into the Radius control plane in my connected environment.***
+
+##### **Summary**
+
+Today, the Application RP container dynamically downloads the Terraform binary from hashicorp.com when a Terraform recipe is executed. This is problematic because Terraform is Radius dependency, not a Radius component. So platform engineers need to be in control of what Terraform is being used. They may use a specific version of Terraform within their organization. They may have specific licensing concerns about Terraform. Or they may simply not want Terraform to be running in their environment.
+
+Therefore, Radius will move to a model where the platform engineer explicitly installs Terraform. This is true for offline installations as well as connected installations.
+
+##### **User Experience 1 – Quick Install**
+
+The platform engineer installs Terraform using the new `rad terraform install` command.
+
+```bash
+$ rad terraform install
+No Terraform release specified, installing the latest from releases.hashicorp.com
+terraform_1.12.2_linux_amd64 installed in the Radius control plane
+```
+
+When this command is executed, Radius:
+
+* Downloads the latest Terraform release from releases.hashicorp.com and validates the checksum
+* Installs the Terraform binary in the appropriate location within the Radius control plane
+* Performs a `terraform init` and reports any output to the platform engineer
+
+Note that `terraform init` will need to be re-executed on recipe execution to initialize to a user-provided backend which happens later.
+
+Changes from today's Radius:
+
+* Terraform is no longer dynamically installed in Application RP
+
+**User Experience 2 – Advanced Install**
+
+The platform engineer uses the same command but with specific URL of the binary and its checksum.
```

Comment:
Is there a need to also allow uploading the binary from a local machine or network share?